pentaho/src/main/java/pt/webdetails/cpf/olap/AbstractOlapUtils.java

```diff
@@ -216,11 +212,53 @@ protected Connection getMdxConnection( String catalog ) {
   }
 
   private Connection getMdxConnection( String catalog, String jndi ) {
+    MondrianCatalog selectedCatalog = getMondrianCatalog( catalog );
+
+    if ( selectedCatalog == null ) {
+      logger.error( "Received catalog '" + cleanLogString( catalog ) + "' doesn't appear to be valid" );
+      return null;
+    }
+
+    if ( !isJNDIValid( jndi ) ) {
+      logger.error( "Received jndi '" + cleanLogString( jndi ) + "' doesn't appear to be valid" );
+      return null;
+    }
+
     String connectStr = "provider=mondrian;dataSource=" + jndi + ";Catalog=" + catalog;
```

Comment:
for consistency should it be `Catalog=" + selectedCatalog.getDefinition();`?

Reply:
I kept the connection string because I wanted to validate the user input without altering the behavior because the `catalog` and `selectedCatalog.getDefinition()` return different things. However, I did have the same concern and ultimately decided to maintain the current version.